packages/components/src/Layout/Layout.component.js

```diff
@@ -54,22 +72,31 @@ function Layout({
 		default:
 			Component = OneColumn;
 	}
+
+	const safeDrawers = getDrawers(getComponent, drawers, components);
+	const safeHeader = Inject.getReactElement(getComponent, header);
+	const safeSubHeader = Inject.getReactElement(getComponent, subHeader);
+	const safeFooter = Inject.getReactElement(getComponent, footer);
+
 	return (
 		<div id={id} className={appCSS}>
-			{header && (
+			{safeHeader && (
 				<header role="banner" className={headerCSS}>
-					{header}
+					{safeHeader}
 				</header>
 			)}
-			{subHeader}
+			{inject('after-header')}
```

Comment:
Do we really need to add those entry points.
IMO the layout should define useful slots (header, footer, main) and stick to it.

(If we remove them, don't forget the .md)